Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp

```diff
@@ -270,10 +270,9 @@ RegisterID* ResolveNode::emitBytecode(BytecodeGenerator& generator, RegisterID*
     
     JSTextPosition divot = m_start + m_ident.length();
     generator.emitExpressionInfo(divot, m_start, divot);
-    RefPtr<RegisterID> scope = generator.emitResolveScope(dst, var);
     RegisterID* finalDest = generator.finalDestination(dst);
     RefPtr<RegisterID> uncheckedResult = generator.newTemporary();
-    generator.emitGetFromScope(uncheckedResult.get(), scope.get(), var, ThrowIfNotFound);
+    generator.emitResolveAndGetFromScope(uncheckedResult.get(), dst, var, ThrowIfNotFound);
```

Comment:
Why are we padding `dst` as the resolved scope here? That looks quite wrong.

Reply:
I think the logic is correct. But, the function name `emitResolveAndGetFromScope` is misleading and it should be `tryEmitResolveAndGetFromScope`. If you check the implementation of this function:
```
RegisterID* BytecodeGenerator::emitResolveAndGetFromScope(RegisterID* dst, RegisterID* resolvedScope, const Variable& variable, ResolveMode resolveMode, bool isPrivate)
{
    switch (variable.offset().kind()) {
    case VarKind::Stack: {
        ASSERT(!isPrivate);
        UNUSED_VARIABLE(isPrivate);
        return emitGetFromScopeWithVarKindStack(dst, variable);
    }
    case VarKind::DirectArgument:
        return emitGetFromScopeWithVarKindDirectArgument(dst, argumentsRegister(), variable);
    case VarKind::Scope:
        return emitGetFromScopeHelper(dst, emitResolveScopeWithVarKindScope(variable), variable, resolveMode);
    case VarKind::Invalid: {
#if USE(JSVALUE64)
        if (!variable.offset().isScope()) {
            OpResolveAndGetFromScope::emit(
                this,
                kill(dst),
                scopeRegister(),
                addConstant(variable.ident()),
                GetPutInfo(resolveMode, resolveType(), InitializationMode::NotInitialization, ecmaMode()),
                localScopeDepth(),
                variable.offset().isScope() ? variable.offset().scopeOffset().offset() : 0);
            return dst;
        }
#endif
        return emitGetFromScopeHelper(dst, emitResolveScopeHelper(resolvedScope, variable), variable, resolveMode);
    }
    }

    RELEASE_ASSERT_NOT_REACHED();
    return nullptr;
}
```
which still need to handle the non-ResolveAndGetFromScope cases. Here, the `dst` is passed to the function as the resolvedScope for clause:
```
return emitGetFromScopeHelper(dst, emitResolveScopeHelper(resolvedScope, variable), variable, resolveMode);
```

Reply:
I believe the similar situation happens in both `emitResolveScope` and `emitGetFromScope`. And they should be `tryEmitResolveScope` and `tryEmitGetFromScope` instead.

Reply:
They should not be prefixed with `try`. Whether or not we actually emit the opcodes, we're logically resolving and getting from a scope. I don't see why we need to pass in the `dst` as the result of the resolve_scope. If we need the scope for later operations, we shouldn't be emitting resolve_and_get_from_scope combo opcode.